demisto/client.py

```diff
@@ -86,8 +88,82 @@ def CreateIncident(self, inc_name, inc_type, inc_severity, inc_owner, inc_labels
         return self.req("POST", "incident", data)
 
     def SearchIncidents(self, page, size, query):
-        data = {'filter': {'page': page, 'size': size, 'query': query, 'sort': [{'field':'id', 'asc': False}]}}
+        data = {'filter': {'page': page, 'size': size,
+                           'query': query, 'sort': [{'field': 'id', 'asc': False}]}}
         r = self.req("POST", "incidents/search", data)
         if r.status_code != 200:
-            raise RuntimeError('Error searching incidents - %d (%s)' % (r.status_code, r.reason))
+            raise RuntimeError(
+                'Error searching incidents - %d (%s)' % (r.status_code, r.reason))
         return r.json()
+
+    def SearchAutomation(self, query):
+        data = {'query':query}
+        r = self.req("POST", "automation/search", data)
+        if r.status_code == requests.codes.ok:
+            rj = r.json()
+            if "scripts" not in rj:
+                raise RuntimeError("Error searching automations - scripts attribute is not found in response")
+            return rj
+        r.raise_for_status()
+
+    def LoadAutomation(self, automation_id):
+        data = {}
+        r = self.req("POST", "automation/load/{}".format(automation_id), data)
+        if r.status_code == requests.codes.ok:
+            return r.json()
+        r.raise_for_status()
+    
+    def SaveAutomation(self, script, filter={"query":""}, save_password=False):
+        filter["query"]="name:{}".format(script["name"])
+        data = {
+            'filter': filter,
+            'script': script,
+            'savePassword': save_password
+        }
+        r = self.req("POST", "automation", data)
+        if r.status_code == requests.codes.ok:
+            return r.json()
+        r.raise_for_status()
+
+    def DeleteAutomation(self, script, filter={"query":""}):
+        data = {
+            'filter': filter,
+            'script': script
+        }
+        r = self.req("POST", "automation/delete", data)
+        if r.status_code == requests.codes.ok:
+            return r.json()
+        r.raise_for_status()
```

Comment:
Why not `raise_for_status` first?

```python
r = self.req("POST", "automation/delete", data)
r.raise_for_status()
return r.json()
```

Reply:
use raise_for_status first, then return (f298ab2).